File: D/Post.m
//
//  Post.m
//  Greip
//
//  Created by Emma Nimstad on 24/9/15.
//  Copyright © 2015 Emma Nimstad. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Post.h"

NSDate *defDate;
NSString *defTitle = @"Godmorgon";
NSString *defText = @"Lorem ipsum lotsa aower hawehr aiw uerh liawekbj raiuwe hralekjnlak text alj aie akwej awue lobi oilb k.";
NSDateFormatter *dateFormatterFromServer, *dateFormatterFromDate;

@implementation Post

+ (void)initialize {
    dateFormatterFromServer = [[NSDateFormatter alloc] init];
    dateFormatterFromDate = [[NSDateFormatter alloc] init];
    
//  Change this to match how date is saved on server
    [dateFormatterFromServer setDateFormat:@"d MMM"];//:NSDateFormatterMediumStyle];
    
//  Change this to how the date should be shown
    [dateFormatterFromDate setDateStyle:NSDateFormatterMediumStyle];
    
    defDate = [NSDate dateWithTimeIntervalSince1970:1450000000];
}

- (id) initWith:(NSString *)title :(NSString *)date :(NSString *)text {
    self.title = title;
    self.text = text;
    
    NSDate *postDate = [dateFormatterFromServer dateFromString:date];
    self.date = [dateFormatterFromDate stringFromDate:postDate];
    
    return self;
}

- (id) initwithDefaults {
    
    _date = [dateFormatterFromDate stringFromDate:defDate];
    _title = defTitle;
    _text = defText;
    
    return self;
}

@end

File: Greip/Post.swift
//
//  Post.swift
//  Greip
//
//  Created by Emma Nimstad on 19/5/16.
//  Copyright © 2016 Emma Nimstad. All rights reserved.
//

import Foundation

/// A single item of the DataFeed.
class Post : NSObject {
	let content: NSAttributedString
	let date, title, author: String

	let dateFormatterFromServer = DateFormatter()
	let dateFormatterFromDate = DateFormatter()

	init(title: String, date: String, content: NSAttributedString, author: String) {
		dateFormatterFromDate.dateStyle = .short
		dateFormatterFromServer.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS"

		if let d = dateFormatterFromServer.date(from: date) {
			self.date = dateFormatterFromDate.string(from: d)
		} else {
			dateFormatterFromServer.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
			if let d = dateFormatterFromServer.date(from: date) {
				self.date = dateFormatterFromDate.string(from: d)
			} else {
				self.date = ""
			}
		}

        self.title = title
        self.content = content
		self.author = author
    }
}
